formatter.go

```diff
@@ -30,16 +35,31 @@ type Formatter interface {
 //
 // It's not exported because it's still using Data in an opinionated way. It's to
 // avoid code duplication between the two default formatters.
-func prefixFieldClashes(data Fields) {
+func prefixFieldClashes(data Fields, showCaller bool, depth int) {
 	if t, ok := data["time"]; ok {
 		data["fields.time"] = t
 	}
-
 	if m, ok := data["msg"]; ok {
 		data["fields.msg"] = m
 	}
-
 	if l, ok := data["level"]; ok {
 		data["fields.level"] = l
 	}
+
+	if showCaller {
+		if _, ok := data["caller"]; ok {
+			data["fields.caller"] = data["caller"]
+		}
+		data["caller"] = caller(depth)
+	}
+}
+
```

Comment:
Depth  hard code into Entry struct. logrus.Entry.log()  (entry.go line: 92) invoke entry.Logger.Hooks.Fire() and entry.Reader(). The depth of  entry.Logger.Hooks.Fire()  different from entry.Reader(). Function caller() not cover the case. I give a patch below, hope it's useful.
[https://github.com/puma252/patchtopull349/blob/master/formatter.go](url)